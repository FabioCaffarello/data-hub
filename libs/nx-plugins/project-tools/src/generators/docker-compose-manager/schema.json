{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "DockerComposeManager",
  "title": "",
  "type": "object",
  "properties": {
    "groupServices": {
      "type": "array",
      "items": {
        "type": "string"
      },
      "description": "Select the group services to include in your compose",
      "x-prompt": {
        "type": "select",
        "message": "Select the group services to include in your compose",
        "items": [
          { "value": "bot", "label": "Bot" },
          { "value": "back-end", "label": "Back-end" },
          { "value": "front-end", "label": "front-end" }
        ],
        "multiple": true
      }
    },
    "coreBusiness": {
      "type": "string",
      "description": "coreBusiness"
    },
    "subjectBusiness": {
      "type": "string",
      "description": "subjectBusiness"
    },
    "ingestors": {
      "type": "string",
      "description": "list the bots name"
    }
  },
  "required": ["groupServices"]
}
