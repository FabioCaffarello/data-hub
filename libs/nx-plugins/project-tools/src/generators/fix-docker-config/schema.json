{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "FixDockerConfig",
  "title": "",
  "type": "object",
  "properties": {
  },
  "required": []
}
