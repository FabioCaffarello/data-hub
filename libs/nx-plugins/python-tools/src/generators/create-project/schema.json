{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "NxPluginsCreateProject",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "type": {
      "type": "string",
      "enum": ["application", "library"],
      "description": "Project type application or library",
      "default": "application",
      "x-prompt": {
        "message": "What project type would you like to use?",
        "type": "list",
        "items": [
          {
            "value": "application",
            "label": "Application"
          },
          {
            "value": "library",
            "label": "Library"
          }
        ]
      }
    },
    "packageName": {
      "type": "string",
      "description": "Python package name",
      "x-prompt": "What package name would you like to use?"
    },
    "moduleName": {
      "type": "string",
      "description": "Python module name"
    },
    "description": {
      "type": "string",
      "description": "Project short description",
      "x-prompt": "What description would you like to use?"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed"
    }
  },
  "required": ["name", "type"]
}
