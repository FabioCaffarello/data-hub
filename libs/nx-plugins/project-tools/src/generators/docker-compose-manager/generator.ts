import {
  formatFiles,
  generateFiles,
  ProjectConfiguration,
  Tree
} from '@nrwl/devkit';
import { readWorkspaceConfig } from '@nrwl/workspace';
import * as fs from 'fs';
import * as path from 'path';
import * as yaml from 'yaml';
import { DockerComposeManagerGeneratorSchema } from './schema';
import { propsMappings, serviceProps, stackMappings } from './stack.services';
import { stackTemplates, networkName } from './template.services';
import { getOwnKeys } from '../../../../utils/json/commands.json'


interface NormalizedSchema extends DockerComposeManagerGeneratorSchema {
  botService: boolean;
  backendService: boolean;
  frontendService: boolean;
  ingestorsObject: {botName: string, botPath: string}[];
  workspacePath: string;
  dockerComposePath: string;
}


const devContainerJsonPath = path.join('.devcontainer', 'devcontainer.json');


function normalizeOptions(tree: Tree, options: DockerComposeManagerGeneratorSchema): NormalizedSchema {
  const { groupServices } = options;
  const workspacePath = getWorkspacePath(tree)
  const dockerComposePath = path.join(workspacePath, 'docker-compose.yml');
  let ingestorsObject: {botName: string, botPath: string}[]

  for (const group of groupServices) {
    const prop = propsMappings[group];
    if (prop) {
      serviceProps[prop] = true;
    } else {
      console.warn(`Unknown group '${group}'`);
    }
  }

  if (serviceProps.botService) {
    ingestorsObject = (options.ingestors ?? '')
    .split(',')
    .map((name) => ({
      botName: `ingestors-${options.coreBusiness}-${options.subjectBusiness}-${name}`,
      botPath: getProjectPath(tree, `ingestors-${options.coreBusiness}-${options.subjectBusiness}-${name}`),
    }));
  }

  return {
    ...options,
    ...serviceProps,
    ingestorsObject,
    workspacePath,
    dockerComposePath
  };
}

function getProjectPath(tree: Tree, projectName: string): string {
  const workspaceConfig = readWorkspaceConfig({
    format: 'nx',
    path: path.join(tree.root, 'workspace.json'),
  });
  const projectConfig: ProjectConfiguration | undefined =
    workspaceConfig.projects[projectName];

  return projectConfig?.root ?? '';
}

function addFiles(tree: Tree, options: NormalizedSchema) {
  const templateOptions = {
    ...options,
    offsetFromRoot: '',
    template: ''
  };
  generateFiles(
    tree,
    path.join(__dirname, 'files'),
    options.workspacePath,
    templateOptions
  );
}

function getWorkspacePath(tree: Tree): string {
  const devContainerJson = JSON.parse(fs.readFileSync(devContainerJsonPath, 'utf-8'));
  const workspaceFolder = devContainerJson['workspaceFolder'];
  if (fs.existsSync(devContainerJsonPath)) {
    if (workspaceFolder.includes(tree.root)) {
      return path.join(tree.root, '..', '..')
    } else {
      return tree.root
    }
  } else {
    return ''
  }
}

function addServices(tree: Tree, options: NormalizedSchema) {
  const dockerComposeContent = tree.read(options.dockerComposePath, 'utf-8');
  const dockerComposeObject = dockerComposeContent
    ? yaml.parse(dockerComposeContent)
    : {};
  const dockerComposeServices = dockerComposeObject.services || {};

  if (options.botService || options.backendService || options.frontendService) {
    // All stacks
    for (const stack of getOwnKeys(stackMappings)) {
      stackMappings[stack].forEach((element) => {
        const stackTemplatesObject = { ...stackTemplates[element] };
        dockerComposeServices[element] = stackTemplatesObject;
      });
    }
    if (options.botService) {
      // Bot Stack
      options.ingestorsObject.forEach((element) => {
        const botServiceObject = {
          build: {
            context: element.botPath,
            dockerfile: 'Dockerfile' // path.join(element.botPath, 'Dockerfile')
          },
          networks: [networkName]
        };
        dockerComposeServices[element.botName] = botServiceObject;
      })
    }
  }
  dockerComposeObject.services = dockerComposeServices;
  const updatedDockerComposeContent = yaml.stringify(dockerComposeObject);
  tree.write(options.dockerComposePath, updatedDockerComposeContent);
}

export default async function (tree: Tree, options: DockerComposeManagerGeneratorSchema) {
  const normalizedOptions = normalizeOptions(tree, options);
  addFiles(tree, normalizedOptions);
  addServices(tree, normalizedOptions);
  await formatFiles(tree);
}
