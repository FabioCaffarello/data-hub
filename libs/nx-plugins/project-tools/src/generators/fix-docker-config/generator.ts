import {
  Tree,
} from '@nrwl/devkit';
import { readJson, updateJson } from '../../../../utils/json/commands.json';
import { FixDockerConfigGeneratorSchema } from './schema';
import { expect } from '@jest/globals';


const filePath = '../.docker/config.json'

export default async function (tree: Tree, options: FixDockerConfigGeneratorSchema) {

  const configData = readJson(filePath)
  delete configData.credsStore;
  updateJson(filePath, configData);
}

export function assertGenerateFiles(tree: Tree) {
  const config = readJson(filePath);
  expect(config.credsStore).toBeUndefined();
}
