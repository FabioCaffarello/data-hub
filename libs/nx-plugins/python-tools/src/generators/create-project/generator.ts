import {
  addProjectConfiguration,
  formatFiles,
  generateFiles,
  getWorkspaceLayout,
  names,
  offsetFromRoot,
  Tree,
} from '@nrwl/devkit';
import * as path from 'path';
import { PyprojectToml } from '@nxlv/python'
import { parse, stringify } from '@iarna/toml';
import { CreateProjectGeneratorSchema } from './schema';
import { getCurrentGitUserName, getCurrentGitUserEmail } from '../../../../utils/git/commands.git'
// import spawn from 'cross-spawn';
import { spawnSync } from 'child_process';


interface NormalizedSchema extends CreateProjectGeneratorSchema {
  projectName: string;
  projectRoot: string;
  projectDirectory: string;
  parsedTags: string[];
  moduleName: string;
  authorName: string;
  authorEmail: string;
}

function normalizeOptions(tree: Tree, options: CreateProjectGeneratorSchema): NormalizedSchema {
  const name = names(options.name).fileName;
  const projectDirectory = options.directory
    ? `${names(options.directory).fileName}/${name}`
    : name;
  const projectName = projectDirectory.replace(new RegExp('/', 'g'), '-');
  const moduleName = options.moduleName ? options.moduleName : projectName.replace(new RegExp('-', 'g'), '_');

  let projectRoot = '';
  if (options.type === 'application') {
    projectRoot = `${getWorkspaceLayout(tree).appsDir}/${projectDirectory}`;
  } else {
    projectRoot = `${getWorkspaceLayout(tree).libsDir}/${projectDirectory}`;
  }
  const parsedTags = options.tags
    ? options.tags.split(',').map((s) => s.trim())
    : [];

  const authorName = getCurrentGitUserName()
  const authorEmail = getCurrentGitUserEmail()

  return {
    ...options,
    description: options.description ?? '',
    projectName,
    projectRoot,
    projectDirectory,
    parsedTags,
    moduleName,
    authorName,
    authorEmail,
  };
}

function addFiles(tree: Tree, options: NormalizedSchema) {
    const templateOptions = {
      ...options,
      ...names(options.name),
      offsetFromRoot: offsetFromRoot(options.projectRoot),
      template: ''
    };
    generateFiles(tree, path.join(__dirname, 'files'), options.projectRoot, templateOptions);
}

function updateRootPyprojectToml(
  tree: Tree,
  normalizedOptions: NormalizedSchema
) {
  if (tree.exists('./pyproject.toml')) {
    const rootPyprojectToml = parse(
      tree.read('pyproject.toml', 'utf-8')
    ) as PyprojectToml;
    rootPyprojectToml.tool.poetry.dependencies[normalizedOptions.packageName] =
      {
        path: normalizedOptions.projectRoot,
        develop: true,
      };
      tree.write('pyproject.toml', stringify(rootPyprojectToml));
  }
}

function updateRootPoetryLock(tree: Tree, normalizedOptions: NormalizedSchema) {
  if (tree.exists('./pyproject.toml')) {
    console.log(`Updating root poetry.lock...`);
    const executable = 'poetry';
    const updateArgs = ['update', normalizedOptions.packageName];
    spawnSync(executable, updateArgs, {
      shell: false,
      stdio: 'inherit',
    });
    return
  }
}

function installSharedPythonCore(normalizedOptions: NormalizedSchema) {
  console.log(
    `Add shared python core to the project ${normalizedOptions.projectName}`
  );
  const executable = 'npx';
  const installArgs = [
    'nx',
    'run',
    `${normalizedOptions.projectName}:add`,
    '--name',
    'shared-python-tools-core',
    '--local',
  ];
  spawnSync(executable, installArgs, {
    shell: false,
    stdio: 'inherit',
  });
  return
}

function installSharedPythonDevelopment(normalizedOptions: NormalizedSchema) {
  console.log(
    `Add shared python development to the project ${normalizedOptions.projectName}`
  );
  const executable = 'npx';
  const installArgs = [
    'nx',
    'run',
    `${normalizedOptions.projectName}:add`,
    '--name',
    'shared-python-tools-development',
    '--local',
    '--group',
    'dev',
  ];
  spawnSync(executable, installArgs, {
    shell: false,
    stdio: 'inherit',
  });
  return
}

async function generator(tree: Tree, options: CreateProjectGeneratorSchema) {
  const normalizedOptions = normalizeOptions(tree, options);
  addProjectConfiguration(
    tree,
    normalizedOptions.projectName,
    {
      root: normalizedOptions.projectRoot,
      projectType: normalizedOptions.type,
      sourceRoot: `${normalizedOptions.projectRoot}/${normalizedOptions.moduleName}`,
      targets: {
        lock: {
          executor: '@nrwl/workspace:run-commands',
          options: {
            command: 'poetry lock --no-update',
            cwd: normalizedOptions.projectRoot,
          },
        },
        add: {
          executor: '@nxlv/python:add',
          options: {},
        },
        update: {
          executor: '@nxlv/python:update',
          options: {},
        },
        remove: {
          executor: '@nxlv/python:remove',
          options: {},
        },
        install: {
          executor: '@nxlv/python:install',
          options: {
            silent: false,
            args: '',
            cacheDir: `.cache/pypoetry`,
            verbose: false,
            debug: false,
          },
        }
      },
      tags: normalizedOptions.parsedTags,
    }
  );
  addFiles(tree, normalizedOptions);
  updateRootPyprojectToml(tree, normalizedOptions);
  await formatFiles(tree);

  return () => {
    installSharedPythonCore(normalizedOptions);
    installSharedPythonDevelopment(normalizedOptions);
    updateRootPoetryLock(tree, normalizedOptions);
  };
}

export default generator;
