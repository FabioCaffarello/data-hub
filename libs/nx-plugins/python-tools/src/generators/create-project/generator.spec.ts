import { spawnSyncMock } from '../../../../utils/mocks/cross-spawn.mock';
import { readProjectConfiguration, Tree } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';

import generator from './generator';
import { CreateProjectGeneratorSchema } from './schema';
import dedent from "dedent";


describe('create-project generator', () => {
  let appTree: Tree;
  const options: CreateProjectGeneratorSchema = {
    name: 'test',
    type: 'application',
    moduleName: null,
    packageName: 'unittest-test-pkg-name',
  };

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace({layout: 'apps-libs'});
  });

  it('should successfully generate a python project', async () => {
    await generator(appTree, options);
    const config = readProjectConfiguration(appTree, 'test');
    expect(config).toMatchSnapshot();

    assertGenerateFiles(appTree, 'apps/test', 'test');
  });

  it('should successfully generate a python project with custom module name', async () => {
    await generator(appTree, { ...options, moduleName: 'mymodule' });
    const config = readProjectConfiguration(appTree, 'test');
    expect(config).toMatchSnapshot();

    assertGenerateFiles(appTree, 'apps/test', 'mymodule');
  });

  it('should successfully generate a python library project', async () => {
    await generator(appTree, { ...options, type: 'library' });
    const config = readProjectConfiguration(appTree, 'test');
    expect(config).toMatchSnapshot();

    assertGenerateFiles(appTree, 'libs/test', 'test');
  });

  it('should generate a python project into a different folder', async () => {
    await generator(appTree, {
      ...options,
      directory: 'shared',
    });
    const config = readProjectConfiguration(appTree, 'shared-test');
    expect(config).toMatchSnapshot();

    assertGenerateFiles(appTree, 'apps/shared/test', 'shared_test');
  });

  it('should generate a python project using description option', async () => {
    await generator(appTree, {
      ...options,
      description: 'My custom description',
    });
    expect(
      appTree.read('apps/test/pyproject.toml').toString()
    ).toMatchSnapshot();
  });

  it('should generate a python project with tags', async () => {
    await generator(appTree, {
      ...options,
      tags: 'python-project, nx, poetry, tox',
    });
    const config = readProjectConfiguration(appTree, 'test');
    expect(config).toMatchSnapshot();
  });

  it('should successfully generate a python library project with root pyproject.toml', async () => {
    appTree.write('pyproject.toml', dedent`
    [tool.poetry]
    name = "unit test"
      [tool.poetry.dependencies]
      python = ">=3.8,<3.10"
    [build-system]
    requires = [ "poetry-core==1.1.0b3" ]
    build-backend = "poetry.core.masonry.api"
    `)

    const callbackTask = await generator(appTree, { ...options, type: 'library' });
    callbackTask();
    const config = readProjectConfiguration(appTree, 'test');
    expect(config).toMatchSnapshot();

    expect(appTree.read('pyproject.toml', 'utf8')).toMatchSnapshot()
    expect(spawnSyncMock).toHaveBeenCalledWith('poetry', ['update', options.packageName], {
      shell: false,
      stdio: 'inherit',
    });
  });
});


function assertGenerateFiles(
  appTree: Tree,
  projectDirectory: string,
  moduleName: string
) {
  expect(appTree.exists(`${projectDirectory}/README.md`)).toBeTruthy();
  expect(
    appTree.read(`${projectDirectory}/README.md`).toString()
  ).toMatchSnapshot();
  expect(appTree.exists(`${projectDirectory}/pyproject.toml`)).toBeTruthy();
  expect(
    appTree.read(`${projectDirectory}/pyproject.toml`).toString()
  ).toMatchSnapshot();
  expect(appTree.exists(`${projectDirectory}/CHANGELOG.md`)).toBeTruthy();
  expect(
    appTree.read(`${projectDirectory}/CHANGELOG.md`).toString()
  ).toMatchSnapshot();
  expect(
    appTree.exists(`${projectDirectory}/${moduleName}/__init__.py`)
  ).toBeTruthy();
  expect(
    appTree.read(`${projectDirectory}/${moduleName}/__init__.py`).toString()
  ).toMatchSnapshot();
}
