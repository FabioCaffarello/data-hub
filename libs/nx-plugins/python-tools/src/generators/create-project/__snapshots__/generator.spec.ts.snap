// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-project generator should generate a python project into a different folder 1`] = `
Object {
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "name": "shared-test",
  "projectType": "application",
  "root": "apps/shared/test",
  "sourceRoot": "apps/shared/test/shared_test",
  "tags": Array [],
  "targets": Object {
    "add": Object {
      "executor": "@nxlv/python:add",
      "options": Object {},
    },
    "install": Object {
      "executor": "@nxlv/python:install",
      "options": Object {
        "args": "",
        "cacheDir": ".cache/pypoetry",
        "debug": false,
        "silent": false,
        "verbose": false,
      },
    },
    "lock": Object {
      "executor": "@nrwl/workspace:run-commands",
      "options": Object {
        "command": "poetry lock --no-update",
        "cwd": "apps/shared/test",
      },
    },
    "remove": Object {
      "executor": "@nxlv/python:remove",
      "options": Object {},
    },
    "update": Object {
      "executor": "@nxlv/python:update",
      "options": Object {},
    },
  },
}
`;

exports[`create-project generator should generate a python project into a different folder 2`] = `
"# shared-test

## About

Project description here.

[API Documentation]()

## [Change log](CHANGELOG.md)
"
`;

exports[`create-project generator should generate a python project into a different folder 3`] = `
"[tool.coverage.run]
branch = true
source = [ \\"shared_test\\" ]

[tool.coverage.report]
exclude_lines = ['if TYPE_CHECKING:']
show_missing = true

[tool.pytest.ini_options]
addopts = \\"--cov --cov-fail-under=100 --cov-report html:'../../../coverage/apps/shared/test/html' --cov-report xml:'../../../coverage/apps/shared/test/coverage.xml' --junitxml='../../../reports/apps/shared/test/unittests/junit.xml' --html='../../../reports/apps/shared/test/unittests/html/index.html'\\"

[tool.poetry]
name = \\"unittest-test-pkg-name\\"
version = \\"1.0.0\\"
description = \\"\\"
authors = [ ]
license = 'Proprietary'
readme = 'README.md'

  [[tool.poetry.packages]]
  include = \\"shared_test\\"

  [tool.poetry.dependencies]
  python = \\">=3.8,<3.10\\"

[build-system]
requires = [\\"poetry-core==1.1.0b3\\"]
build-backend = \\"poetry.core.masonry.api\\"
"
`;

exports[`create-project generator should generate a python project into a different folder 4`] = `
"# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## Unreleased
"
`;

exports[`create-project generator should generate a python project into a different folder 5`] = `
"\\"\\"\\"\\"\\"\\"
"
`;

exports[`create-project generator should generate a python project using description option 1`] = `
"[tool.coverage.run]
branch = true
source = [ \\"test\\" ]

[tool.coverage.report]
exclude_lines = ['if TYPE_CHECKING:']
show_missing = true

[tool.pytest.ini_options]
addopts = \\"--cov --cov-fail-under=100 --cov-report html:'../../coverage/apps/test/html' --cov-report xml:'../../coverage/apps/test/coverage.xml' --junitxml='../../reports/apps/test/unittests/junit.xml' --html='../../reports/apps/test/unittests/html/index.html'\\"

[tool.poetry]
name = \\"unittest-test-pkg-name\\"
version = \\"1.0.0\\"
description = \\"My custom description\\"
authors = [ ]
license = 'Proprietary'
readme = 'README.md'

  [[tool.poetry.packages]]
  include = \\"test\\"

  [tool.poetry.dependencies]
  python = \\">=3.8,<3.10\\"

[build-system]
requires = [\\"poetry-core==1.1.0b3\\"]
build-backend = \\"poetry.core.masonry.api\\"
"
`;

exports[`create-project generator should generate a python project with tags 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "test",
  "projectType": "application",
  "root": "apps/test",
  "sourceRoot": "apps/test/test",
  "tags": Array [
    "python-project",
    "nx",
    "poetry",
    "tox",
  ],
  "targets": Object {
    "add": Object {
      "executor": "@nxlv/python:add",
      "options": Object {},
    },
    "install": Object {
      "executor": "@nxlv/python:install",
      "options": Object {
        "args": "",
        "cacheDir": ".cache/pypoetry",
        "debug": false,
        "silent": false,
        "verbose": false,
      },
    },
    "lock": Object {
      "executor": "@nrwl/workspace:run-commands",
      "options": Object {
        "command": "poetry lock --no-update",
        "cwd": "apps/test",
      },
    },
    "remove": Object {
      "executor": "@nxlv/python:remove",
      "options": Object {},
    },
    "update": Object {
      "executor": "@nxlv/python:update",
      "options": Object {},
    },
  },
}
`;

exports[`create-project generator should successfully generate a python library project 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "test",
  "projectType": "library",
  "root": "libs/test",
  "sourceRoot": "libs/test/test",
  "tags": Array [],
  "targets": Object {
    "add": Object {
      "executor": "@nxlv/python:add",
      "options": Object {},
    },
    "install": Object {
      "executor": "@nxlv/python:install",
      "options": Object {
        "args": "",
        "cacheDir": ".cache/pypoetry",
        "debug": false,
        "silent": false,
        "verbose": false,
      },
    },
    "lock": Object {
      "executor": "@nrwl/workspace:run-commands",
      "options": Object {
        "command": "poetry lock --no-update",
        "cwd": "libs/test",
      },
    },
    "remove": Object {
      "executor": "@nxlv/python:remove",
      "options": Object {},
    },
    "update": Object {
      "executor": "@nxlv/python:update",
      "options": Object {},
    },
  },
}
`;

exports[`create-project generator should successfully generate a python library project 2`] = `
"# test

## About

Project description here.

[API Documentation]()

## [Change log](CHANGELOG.md)
"
`;

exports[`create-project generator should successfully generate a python library project 3`] = `
"[tool.coverage.run]
branch = true
source = [ \\"test\\" ]

[tool.coverage.report]
exclude_lines = ['if TYPE_CHECKING:']
show_missing = true

[tool.pytest.ini_options]
addopts = \\"--cov --cov-fail-under=100 --cov-report html:'../../coverage/libs/test/html' --cov-report xml:'../../coverage/libs/test/coverage.xml' --junitxml='../../reports/libs/test/unittests/junit.xml' --html='../../reports/libs/test/unittests/html/index.html'\\"

[tool.poetry]
name = \\"unittest-test-pkg-name\\"
version = \\"1.0.0\\"
description = \\"\\"
authors = [ ]
license = 'Proprietary'
readme = 'README.md'

  [[tool.poetry.packages]]
  include = \\"test\\"

  [tool.poetry.dependencies]
  python = \\">=3.8,<3.10\\"

[build-system]
requires = [\\"poetry-core==1.1.0b3\\"]
build-backend = \\"poetry.core.masonry.api\\"
"
`;

exports[`create-project generator should successfully generate a python library project 4`] = `
"# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## Unreleased
"
`;

exports[`create-project generator should successfully generate a python library project 5`] = `
"\\"\\"\\"\\"\\"\\"
"
`;

exports[`create-project generator should successfully generate a python library project with root pyproject.toml 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "test",
  "projectType": "library",
  "root": "libs/test",
  "sourceRoot": "libs/test/test",
  "tags": Array [],
  "targets": Object {
    "add": Object {
      "executor": "@nxlv/python:add",
      "options": Object {},
    },
    "install": Object {
      "executor": "@nxlv/python:install",
      "options": Object {
        "args": "",
        "cacheDir": ".cache/pypoetry",
        "debug": false,
        "silent": false,
        "verbose": false,
      },
    },
    "lock": Object {
      "executor": "@nrwl/workspace:run-commands",
      "options": Object {
        "command": "poetry lock --no-update",
        "cwd": "libs/test",
      },
    },
    "remove": Object {
      "executor": "@nxlv/python:remove",
      "options": Object {},
    },
    "update": Object {
      "executor": "@nxlv/python:update",
      "options": Object {},
    },
  },
}
`;

exports[`create-project generator should successfully generate a python library project with root pyproject.toml 2`] = `
"[tool.poetry]
name = \\"unit test\\"

  [tool.poetry.dependencies]
  python = \\">=3.8,<3.10\\"

    [tool.poetry.dependencies.unittest-test-pkg-name]
    path = \\"libs/test\\"
    develop = true

[build-system]
requires = [ \\"poetry-core==1.1.0b3\\" ]
build-backend = \\"poetry.core.masonry.api\\"
"
`;

exports[`create-project generator should successfully generate a python project 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "test",
  "projectType": "application",
  "root": "apps/test",
  "sourceRoot": "apps/test/test",
  "tags": Array [],
  "targets": Object {
    "add": Object {
      "executor": "@nxlv/python:add",
      "options": Object {},
    },
    "install": Object {
      "executor": "@nxlv/python:install",
      "options": Object {
        "args": "",
        "cacheDir": ".cache/pypoetry",
        "debug": false,
        "silent": false,
        "verbose": false,
      },
    },
    "lock": Object {
      "executor": "@nrwl/workspace:run-commands",
      "options": Object {
        "command": "poetry lock --no-update",
        "cwd": "apps/test",
      },
    },
    "remove": Object {
      "executor": "@nxlv/python:remove",
      "options": Object {},
    },
    "update": Object {
      "executor": "@nxlv/python:update",
      "options": Object {},
    },
  },
}
`;

exports[`create-project generator should successfully generate a python project 2`] = `
"# test

## About

Project description here.

[API Documentation]()

## [Change log](CHANGELOG.md)
"
`;

exports[`create-project generator should successfully generate a python project 3`] = `
"[tool.coverage.run]
branch = true
source = [ \\"test\\" ]

[tool.coverage.report]
exclude_lines = ['if TYPE_CHECKING:']
show_missing = true

[tool.pytest.ini_options]
addopts = \\"--cov --cov-fail-under=100 --cov-report html:'../../coverage/apps/test/html' --cov-report xml:'../../coverage/apps/test/coverage.xml' --junitxml='../../reports/apps/test/unittests/junit.xml' --html='../../reports/apps/test/unittests/html/index.html'\\"

[tool.poetry]
name = \\"unittest-test-pkg-name\\"
version = \\"1.0.0\\"
description = \\"\\"
authors = [ ]
license = 'Proprietary'
readme = 'README.md'

  [[tool.poetry.packages]]
  include = \\"test\\"

  [tool.poetry.dependencies]
  python = \\">=3.8,<3.10\\"

[build-system]
requires = [\\"poetry-core==1.1.0b3\\"]
build-backend = \\"poetry.core.masonry.api\\"
"
`;

exports[`create-project generator should successfully generate a python project 4`] = `
"# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## Unreleased
"
`;

exports[`create-project generator should successfully generate a python project 5`] = `
"\\"\\"\\"\\"\\"\\"
"
`;

exports[`create-project generator should successfully generate a python project with custom module name 1`] = `
Object {
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "test",
  "projectType": "application",
  "root": "apps/test",
  "sourceRoot": "apps/test/mymodule",
  "tags": Array [],
  "targets": Object {
    "add": Object {
      "executor": "@nxlv/python:add",
      "options": Object {},
    },
    "install": Object {
      "executor": "@nxlv/python:install",
      "options": Object {
        "args": "",
        "cacheDir": ".cache/pypoetry",
        "debug": false,
        "silent": false,
        "verbose": false,
      },
    },
    "lock": Object {
      "executor": "@nrwl/workspace:run-commands",
      "options": Object {
        "command": "poetry lock --no-update",
        "cwd": "apps/test",
      },
    },
    "remove": Object {
      "executor": "@nxlv/python:remove",
      "options": Object {},
    },
    "update": Object {
      "executor": "@nxlv/python:update",
      "options": Object {},
    },
  },
}
`;

exports[`create-project generator should successfully generate a python project with custom module name 2`] = `
"# test

## About

Project description here.

[API Documentation]()

## [Change log](CHANGELOG.md)
"
`;

exports[`create-project generator should successfully generate a python project with custom module name 3`] = `
"[tool.coverage.run]
branch = true
source = [ \\"mymodule\\" ]

[tool.coverage.report]
exclude_lines = ['if TYPE_CHECKING:']
show_missing = true

[tool.pytest.ini_options]
addopts = \\"--cov --cov-fail-under=100 --cov-report html:'../../coverage/apps/test/html' --cov-report xml:'../../coverage/apps/test/coverage.xml' --junitxml='../../reports/apps/test/unittests/junit.xml' --html='../../reports/apps/test/unittests/html/index.html'\\"

[tool.poetry]
name = \\"unittest-test-pkg-name\\"
version = \\"1.0.0\\"
description = \\"\\"
authors = [ ]
license = 'Proprietary'
readme = 'README.md'

  [[tool.poetry.packages]]
  include = \\"mymodule\\"

  [tool.poetry.dependencies]
  python = \\">=3.8,<3.10\\"

[build-system]
requires = [\\"poetry-core==1.1.0b3\\"]
build-backend = \\"poetry.core.masonry.api\\"
"
`;

exports[`create-project generator should successfully generate a python project with custom module name 4`] = `
"# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## Unreleased
"
`;

exports[`create-project generator should successfully generate a python project with custom module name 5`] = `
"\\"\\"\\"\\"\\"\\"
"
`;
